### PR TITLE
Add Java's status to the support matrix

### PR DESCRIPTION
**Pull request by tylerbenson**
Also renamed column to clarify name.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,14 +62,14 @@ The table below captures the state of various features and their levels of suppo
...
-| Custom context propagation |      |        |      | N/A  | N/A  |      |
-| Span link propagation      |      |        |      | N/A  | N/A  |      |
-| Semantic conventions       |      |        |      | N/A  | N/A  |      |
-| - Trace General            |      |        |      | N/A  | N/A  |      |
-| - Trace Incoming           |      |        |      | N/A  | N/A  |      |
-| - Trace Outgoing           |      |        |      | N/A  | N/A  |      |
-| - Metrics                  |      |        |      | N/A  | N/A  |      |
-| Auto instrumentation       |      |        |      | N/A  | N/A  |      |
+| Custom context propagation |      |        |  -   | N/A  | N/A  |      |
+| X-Ray Env Var Span Link    |      |        |  -   | N/A  | N/A  |      |
```
Comment (codeboten) [inline README.md:66]: Please rename the header in the features section as well

File: README.md [hunk 2]
```diff
@@ -62,14 +62,14 @@ The table below captures the state of various features and their levels of suppo
...
+| Semantic conventions       |      |        |  +   | N/A  | N/A  |      |
```
Comment (codeboten) [inline README.md]: Can the sections under semantic conventions be marked as done before the reviews of the spec (and any changes necessary) are completed?

Comment (tylerbenson) [reply README.md]: Is this better?

**Review by codeboten — APPROVED** [review]
Thanks @tylerbenson

**Pull request merged**